File: exercise/scala/day08/src/test/scala/ToyProductionServiceSpec.scala
import domain.Toy
import domain.Toy.State.*
import doubles.InMemoryToyRepository
import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers
import service.ToyProductionService

class ToyProductionServiceSpec extends AnyFunSuite with Matchers {
  private val toyName = "Train"

  test("assignToyToElf should pass the item in production") {
    val repository = new InMemoryToyRepository()
    val service = new ToyProductionService(repository)
    repository.save(new Toy(toyName, UNASSIGNED))

    service.assignToyToElf(toyName)

    repository.findByName(toyName).map(_.state) shouldBe Some(IN_PRODUCTION)
  }
}

File: exercise/ts/day08/tests/toyProductionService.spec.ts
import {InMemoryToyRepository} from "./doubles/inMemoryToyRepository";
import {ToyProductionService} from "../src/services/toyProductionService";
import {Toy} from "../src/domain/toy";

describe('ToyProductionService', () => {
    const TOY_NAME = 'Train';

    it('assignToyToElfShouldPassTheItemInProduction', () => {
        const repository = new InMemoryToyRepository();
        const service = new ToyProductionService(repository);
        repository.save(new Toy(TOY_NAME, Toy.State.UNASSIGNED));

        service.assignToyToElf(TOY_NAME);

        const toy = repository.findByName(TOY_NAME);
        expect(toy?.getState()).toBe(Toy.State.IN_PRODUCTION);
    });
});
